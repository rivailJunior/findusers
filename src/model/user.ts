export const UserValue: User = {
    avatar_url: "https://avatars.githubusercontent.com/u/5783143?v=4",
    bio: "Full Stack developer, javascript lover and an open source enthusiastic.",
    blog: "",
    company: "Qtarkuss",
    created_at: "2013-10-26T19:17:03Z",
    email: "rivail.rj@gmail.com",
    events_url: "https://api.github.com/users/rivailJunior/events{/privacy}",
    followers: 4,
    followers_url: "https://api.github.com/users/rivailJunior/followers",
    following: 2,
    following_url: "https://api.github.com/users/rivailJunior/following{/other_user}",
    gists_url: "https://api.github.com/users/rivailJunior/gists{/gist_id}",
    gravatar_id: "",
    hireable: true,
    html_url: "https://github.com/rivailJunior",
    id: 0,
    location: "Brasil",
    login: "rivailJunior",
    name: "Rivail Santos",
    node_id: "MDQ6VXNlcjU3ODMxNDM=",
    organizations_url: "https://api.github.com/users/rivailJunior/orgs",
    public_gists: 0,
    public_repos: 12,
    received_events_url: "https://api.github.com/users/rivailJunior/received_events",
    repos_url: "https://api.github.com/users/rivailJunior/repos",
    site_admin: false,
    starred_url: "https://api.github.com/users/rivailJunior/starred{/owner}{/repo}",
    subscriptions_url: "https://api.github.com/users/rivailJunior/subscriptions",
    twitter_username: null,
    type: "User",
    updated_at: "2021-04-25T03:20:04Z",
    url: "https://api.github.com/users/rivailJunior"
}

export type User = {
    avatar_url: string
    bio: string
    blog: string
    company: string
    created_at: string
    email: string
    events_url: string
    followers: number
    followers_url: string
    following: number
    following_url: string
    gists_url: string
    gravatar_id: string
    hireable: boolean
    html_url: string
    id: number
    location: string
    login: string
    name: string
    node_id: string
    organizations_url: string
    public_gists: number
    public_repos: number
    received_events_url: string
    repos_url: string
    site_admin: false
    starred_url: string
    subscriptions_url: string
    twitter_username: null
    type: string
    updated_at: string
    url: string
}