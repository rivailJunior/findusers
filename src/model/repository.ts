import { User, UserValue } from './user';
export type Repository = {
    archive_url: string
    archived: boolean
    assignees_url: string
    blobs_url: string
    branches_url: string
    clone_url: string
    collaborators_url: string
    comments_url: string
    commits_url: string
    compare_url: string
    contents_url: string
    contributors_url: string
    created_at: string
    default_branch: string
    deployments_url: string
    description: any
    disabled: boolean
    downloads_url: string
    events_url: string
    fork: number
    forks: number
    forks_count: number
    forks_url: string
    full_name: string
    git_commits_url: string
    git_refs_url: string
    git_tags_url: string
    git_url: string
    has_downloads: boolean
    has_issues: boolean
    has_pages: boolean
    has_projects: boolean
    has_wiki: boolean
    homepage: any
    hooks_url: string
    html_url: string
    id: number
    issue_comment_url: string
    issue_events_url: string
    issues_url: string
    keys_url: string
    labels_url: string
    language: string
    languages_url: string
    license: { key: string, name: string, spdx_id: string, url: string }
    merges_url: string
    milestones_url: string
    mirror_url: any
    name: string
    network_count: number
    node_id: string
    notifications_url: string
    open_issues: number
    open_issues_count: number
    owner: User
    permissions: Object
    private: false
    pulls_url: string
    pushed_at: string
    releases_url: string
    size: number
    ssh_url: string
    stargazers_count: number
    stargazers_url: string
    statuses_url: string
    subscribers_count: number
    subscribers_url: string
    subscription_url: string
    svn_url: string
    tags_url: string
    teams_url: string
    temp_clone_token: string
    trees_url: string
    updated_at: string
    url: string
    watchers: number
    watchers_count: number
}

export const repositoryMockValue = {
    archive_url: "https://api.github.com/repos/rivailJunior/moveon/{archive_format}{/ref}",
    archived: false,
    assignees_url: "https://api.github.com/repos/rivailJunior/moveon/assignees{/user}",
    blobs_url: "https://api.github.com/repos/rivailJunior/moveon/git/blobs{/sha}",
    branches_url: "https://api.github.com/repos/rivailJunior/moveon/branches{/branch}",
    clone_url: "https://github.com/rivailJunior/moveon.git",
    collaborators_url: "https://api.github.com/repos/rivailJunior/moveon/collaborators{/collaborator}",
    comments_url: "https://api.github.com/repos/rivailJunior/moveon/comments{/number}",
    commits_url: "https://api.github.com/repos/rivailJunior/moveon/commits{/sha}",
    compare_url: "https://api.github.com/repos/rivailJunior/moveon/compare/{base}...{head}",
    contents_url: "https://api.github.com/repos/rivailJunior/moveon/contents/{+path}",
    contributors_url: "https://api.github.com/repos/rivailJunior/moveon/contributors",
    created_at: "2016-06-02T19:09:35Z",
    default_branch: "master",
    deployments_url: "https://api.github.com/repos/rivailJunior/moveon/deployments",
    description: null,
    disabled: false,
    downloads_url: "https://api.github.com/repos/rivailJunior/moveon/downloads",
    events_url: "https://api.github.com/repos/rivailJunior/moveon/events",
    fork: false,
    forks: 0,
    forks_count: 0,
    forks_url: "https://api.github.com/repos/rivailJunior/moveon/forks",
    full_name: "rivailJunior/moveon",
    git_commits_url: "https://api.github.com/repos/rivailJunior/moveon/git/commits{/sha}",
    git_refs_url: "https://api.github.com/repos/rivailJunior/moveon/git/refs{/sha}",
    git_tags_url: "https://api.github.com/repos/rivailJunior/moveon/git/tags{/sha}",
    git_url: "git://github.com/rivailJunior/moveon.git",
    has_downloads: true,
    has_issues: true,
    has_pages: false,
    has_projects: true,
    has_wiki: true,
    homepage: null,
    hooks_url: "https://api.github.com/repos/rivailJunior/moveon/hooks",
    html_url: "https://github.com/rivailJunior/moveon",
    id: 60290361,
    issue_comment_url: "https://api.github.com/repos/rivailJunior/moveon/issues/comments{/number}",
    issue_events_url: "https://api.github.com/repos/rivailJunior/moveon/issues/events{/number}",
    issues_url: "https://api.github.com/repos/rivailJunior/moveon/issues{/number}",
    keys_url: "https://api.github.com/repos/rivailJunior/moveon/keys{/key_id}",
    labels_url: "https://api.github.com/repos/rivailJunior/moveon/labels{/name}",
    language: "HTML",
    languages_url: "https://api.github.com/repos/rivailJunior/moveon/languages",
    license: { key: "mit", name: "MIT License", spdx_id: "MIT", url: "https://api.github.com/licenses/mit" },
    merges_url: "https://api.github.com/repos/rivailJunior/moveon/merges",
    milestones_url: "https://api.github.com/repos/rivailJunior/moveon/milestones{/number}",
    mirror_url: null,
    name: "moveon",
    network_count: 0,
    node_id: "MDEwOlJlcG9zaXRvcnk2MDI5MDM2MQ==",
    notifications_url: "https://api.github.com/repos/rivailJunior/moveon/notifications{?since,all,participating}",
    open_issues: 7,
    open_issues_count: 7,
    owner: UserValue,
    permissions: { admin: true, push: true, pull: true },
    private: false,
    pulls_url: "https://api.github.com/repos/rivailJunior/moveon/pulls{/number}",
    pushed_at: "2016-07-19T19:56:10Z",
    releases_url: "https://api.github.com/repos/rivailJunior/moveon/releases{/id}",
    size: 14671,
    ssh_url: "git@github.com:rivailJunior/moveon.git",
    stargazers_count: 1,
    stargazers_url: "https://api.github.com/repos/rivailJunior/moveon/stargazers",
    statuses_url: "https://api.github.com/repos/rivailJunior/moveon/statuses/{sha}",
    subscribers_count: 5,
    subscribers_url: "https://api.github.com/repos/rivailJunior/moveon/subscribers",
    subscription_url: "https://api.github.com/repos/rivailJunior/moveon/subscription",
    svn_url: "https://github.com/rivailJunior/moveon",
    tags_url: "https://api.github.com/repos/rivailJunior/moveon/tags",
    teams_url: "https://api.github.com/repos/rivailJunior/moveon/teams",
    temp_clone_token: "",
    trees_url: "https://api.github.com/repos/rivailJunior/moveon/git/trees{/sha}",
    updated_at: "2016-09-20T14:47:58Z",
    url: "https://api.github.com/repos/rivailJunior/moveon",
    watchers: 1,
    watchers_count: 1
}