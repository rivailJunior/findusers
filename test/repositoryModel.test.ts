
import { repositoryMockValue } from '../src/model/repository';

import { cleanup } from '@testing-library/react'

import '@testing-library/jest-dom/extend-expect'

afterEach(cleanup)

describe('User types and Value', () => {
    test('UserValues snapshot', () => {
        expect(repositoryMockValue).toMatchSnapshot({
            id: repositoryMockValue.id
        })
    })
})