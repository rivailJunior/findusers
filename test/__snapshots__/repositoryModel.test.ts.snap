// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User types and Value UserValues snapshot 1`] = `
Object {
  "archive_url": "https://api.github.com/repos/rivailJunior/moveon/{archive_format}{/ref}",
  "archived": false,
  "assignees_url": "https://api.github.com/repos/rivailJunior/moveon/assignees{/user}",
  "blobs_url": "https://api.github.com/repos/rivailJunior/moveon/git/blobs{/sha}",
  "branches_url": "https://api.github.com/repos/rivailJunior/moveon/branches{/branch}",
  "clone_url": "https://github.com/rivailJunior/moveon.git",
  "collaborators_url": "https://api.github.com/repos/rivailJunior/moveon/collaborators{/collaborator}",
  "comments_url": "https://api.github.com/repos/rivailJunior/moveon/comments{/number}",
  "commits_url": "https://api.github.com/repos/rivailJunior/moveon/commits{/sha}",
  "compare_url": "https://api.github.com/repos/rivailJunior/moveon/compare/{base}...{head}",
  "contents_url": "https://api.github.com/repos/rivailJunior/moveon/contents/{+path}",
  "contributors_url": "https://api.github.com/repos/rivailJunior/moveon/contributors",
  "created_at": "2016-06-02T19:09:35Z",
  "default_branch": "master",
  "deployments_url": "https://api.github.com/repos/rivailJunior/moveon/deployments",
  "description": null,
  "disabled": false,
  "downloads_url": "https://api.github.com/repos/rivailJunior/moveon/downloads",
  "events_url": "https://api.github.com/repos/rivailJunior/moveon/events",
  "fork": false,
  "forks": 0,
  "forks_count": 0,
  "forks_url": "https://api.github.com/repos/rivailJunior/moveon/forks",
  "full_name": "rivailJunior/moveon",
  "git_commits_url": "https://api.github.com/repos/rivailJunior/moveon/git/commits{/sha}",
  "git_refs_url": "https://api.github.com/repos/rivailJunior/moveon/git/refs{/sha}",
  "git_tags_url": "https://api.github.com/repos/rivailJunior/moveon/git/tags{/sha}",
  "git_url": "git://github.com/rivailJunior/moveon.git",
  "has_downloads": true,
  "has_issues": true,
  "has_pages": false,
  "has_projects": true,
  "has_wiki": true,
  "homepage": null,
  "hooks_url": "https://api.github.com/repos/rivailJunior/moveon/hooks",
  "html_url": "https://github.com/rivailJunior/moveon",
  "id": 60290361,
  "issue_comment_url": "https://api.github.com/repos/rivailJunior/moveon/issues/comments{/number}",
  "issue_events_url": "https://api.github.com/repos/rivailJunior/moveon/issues/events{/number}",
  "issues_url": "https://api.github.com/repos/rivailJunior/moveon/issues{/number}",
  "keys_url": "https://api.github.com/repos/rivailJunior/moveon/keys{/key_id}",
  "labels_url": "https://api.github.com/repos/rivailJunior/moveon/labels{/name}",
  "language": "HTML",
  "languages_url": "https://api.github.com/repos/rivailJunior/moveon/languages",
  "license": Object {
    "key": "mit",
    "name": "MIT License",
    "spdx_id": "MIT",
    "url": "https://api.github.com/licenses/mit",
  },
  "merges_url": "https://api.github.com/repos/rivailJunior/moveon/merges",
  "milestones_url": "https://api.github.com/repos/rivailJunior/moveon/milestones{/number}",
  "mirror_url": null,
  "name": "moveon",
  "network_count": 0,
  "node_id": "MDEwOlJlcG9zaXRvcnk2MDI5MDM2MQ==",
  "notifications_url": "https://api.github.com/repos/rivailJunior/moveon/notifications{?since,all,participating}",
  "open_issues": 7,
  "open_issues_count": 7,
  "owner": Object {
    "avatar_url": "https://avatars.githubusercontent.com/u/5783143?v=4",
    "bio": "Full Stack developer, javascript lover and an open source enthusiastic.",
    "blog": "",
    "company": "Qtarkuss",
    "created_at": "2013-10-26T19:17:03Z",
    "email": "rivail.rj@gmail.com",
    "events_url": "https://api.github.com/users/rivailJunior/events{/privacy}",
    "followers": 4,
    "followers_url": "https://api.github.com/users/rivailJunior/followers",
    "following": 2,
    "following_url": "https://api.github.com/users/rivailJunior/following{/other_user}",
    "gists_url": "https://api.github.com/users/rivailJunior/gists{/gist_id}",
    "gravatar_id": "",
    "hireable": true,
    "html_url": "https://github.com/rivailJunior",
    "id": 0,
    "location": "Brasil",
    "login": "rivailJunior",
    "name": "Rivail Santos",
    "node_id": "MDQ6VXNlcjU3ODMxNDM=",
    "organizations_url": "https://api.github.com/users/rivailJunior/orgs",
    "public_gists": 0,
    "public_repos": 12,
    "received_events_url": "https://api.github.com/users/rivailJunior/received_events",
    "repos_url": "https://api.github.com/users/rivailJunior/repos",
    "site_admin": false,
    "starred_url": "https://api.github.com/users/rivailJunior/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/rivailJunior/subscriptions",
    "twitter_username": null,
    "type": "User",
    "updated_at": "2021-04-25T03:20:04Z",
    "url": "https://api.github.com/users/rivailJunior",
  },
  "permissions": Object {
    "admin": true,
    "pull": true,
    "push": true,
  },
  "private": false,
  "pulls_url": "https://api.github.com/repos/rivailJunior/moveon/pulls{/number}",
  "pushed_at": "2016-07-19T19:56:10Z",
  "releases_url": "https://api.github.com/repos/rivailJunior/moveon/releases{/id}",
  "size": 14671,
  "ssh_url": "git@github.com:rivailJunior/moveon.git",
  "stargazers_count": 1,
  "stargazers_url": "https://api.github.com/repos/rivailJunior/moveon/stargazers",
  "statuses_url": "https://api.github.com/repos/rivailJunior/moveon/statuses/{sha}",
  "subscribers_count": 5,
  "subscribers_url": "https://api.github.com/repos/rivailJunior/moveon/subscribers",
  "subscription_url": "https://api.github.com/repos/rivailJunior/moveon/subscription",
  "svn_url": "https://github.com/rivailJunior/moveon",
  "tags_url": "https://api.github.com/repos/rivailJunior/moveon/tags",
  "teams_url": "https://api.github.com/repos/rivailJunior/moveon/teams",
  "temp_clone_token": "",
  "trees_url": "https://api.github.com/repos/rivailJunior/moveon/git/trees{/sha}",
  "updated_at": "2016-09-20T14:47:58Z",
  "url": "https://api.github.com/repos/rivailJunior/moveon",
  "watchers": 1,
  "watchers_count": 1,
}
`;
